README.md

```diff
@@ -22,59 +22,56 @@ Created something awesome with simplex-noise? Let me know so I can add it to the
 
 ```npm i -S simplex-noise```
 
-## Usage
+## Usage Examples
+
+### ES Module Import
 
 ```javascript
-// when using es modules
-import SimplexNoise from 'simplex-noise';
-// when using commonjs
-const {SimplexNoise} = require('simplex-noise');
+// when using ES modules
+import { noiseFunction2D } from 'simplex-noise';
 ```
 
-By default simplex-noise.js will use Math.random() to seed the noise.
+### CommonJS Require
+
 ```javascript
-// initializing a new simplex instance
-// do this only once as it is relatively expensive
-const simplex = new SimplexNoise(),
-    value2d = simplex.noise2D(x, y),
-    value3d = simplex.noise3D(x, y, z),
-    value4d = simplex.noise4D(x, y, z, w);
+const { noiseFunction2D } = require('simplex-noise');
 ```
 
-You can also pass in a seed string which will then be used to initialize
-the noise using the built in alea PRNG.
-```javascript
-const simplex = new SimplexNoise('seed'),
-    value2d = simplex.noise2D(x, y),
-    sameSeed = new SimplexNoise('seed'),
-    differentSeed = new SimplexNoise('different seed');
+### 2D
 
-sameSeed.noise2D(x, y) === value2d
-differentSeed.noise2D(x, y) !== value2d
+```
+const noise2D = noiseFunction2D();
+console.log(noise2D(x, y));
 ```
 
-You can also pass an alternative random function to the constructor that is
-used to build the permutation table.
-This can be used with a custom pseudo random number generator:
+### 3D
 
-```javascript
-const random = new Alea(seed),
-    simplex = new SimplexNoise(random),
-    value2d = simplex.noise2D(x, y);
+```
+const noise3D = noiseFunction3D();
+console.log(noise3D(x, y, z));
 ```
 
-The ALEA PRNG can be found in the npm package [alea](https://npmjs.org/package/alea).
+### 4D
 
-## node.js
+```
+const noise4D = noiseFunction4D();
+console.log(noise4D(x, y, z, w));
+```
 
-Node.js is also supported, you can install the package using [npm](https://npmjs.org/package/simplex-noise).
+### Using a seed value
 
-```javascript
-const SimplexNoise = require('simplex-noise'),
-    simplex = new SimplexNoise(Math.random),
-    value2d = simplex.noise2D(x, y);
+By default simplex-noise.js will use Math.random() to seed the noise.
+You can pass in a PRNG function to use your own seed value.
+
+```
+import Alea from 'alea';
+const alea = new Alea('seed');
+const noise2D = noiseFunction2D(alea);
```

Comment:
I don't like having the word `function` in the function name. Can we keep the same names we currently have? `noise2D`, `noise3D`, etc. ?

Reply:
```javascript
import { noise2D } from 'simplex-noise';

const noise = noise2D(new Alea('seed'));
```

Reply:
I didn't want to call the factories/constructors noiseND to leave that name up to the user. If you extend the above example to also construct a 3d noise function you'd end up with something like noise2 = noise2D(), noise3 = noise3D() or n2 = noise2D(), n3 = noise3D(). It is also a bit more obvious to me from the name that noiseFunction2D returns a function, whereas noise2D I'd expect to be the noise function itself. It's a very weakly held opinion though.

Reply:
users can choose whatever name they like for the import though. For example:

```javascript
import { noise2D as nz } from 'simplex-noise'

const noise2D = nz(new Alea('seed'))
```

Reply:
another option is maybe `simpleNoiseXD` ?  

```javascript
import { simplexNoise2D } from 'simplex-noise'
```

Reply:
Renamed them to createNoiseND() I hope that is a bit better for you as well. :)